main/src/ca/uwaterloo/flix/language/phase/LoadBytecode.scala

```diff
@@ -105,16 +127,81 @@ object LoadBytecode extends Phase[ExecutableAst.Root, ExecutableAst.Root] {
     val interfaces: Map[Type, List[String]] = generateInterfaceNames(constantsList)
     val loadedInterfaces: Map[Type, Class[_]] = interfaces.map { case (tpe, prefix) =>
       // Use a temporary context with no functions, because the codegen needs the map of interfaces.
-      val bytecode = Codegen.compileFunctionalInterface(Codegen.Context(prefix, List.empty, declarations, interfaces))(tpe)
+      val bytecode = Codegen.compileFunctionalInterface(CodegenHelper.Context(prefix, Nil, declarations,
+        interfaces, Map.empty, Map.empty))(tpe)
       if (flix.options.debug) {
         dump(prefix, bytecode)
       }
       tpe -> loader(prefix, bytecode)
     }.toMap // Despite IDE highlighting, this is actually necessary.
 
-    // 4. Generate and load bytecode.
+
+    // 4. Extract declared enums.
+    val declaredEnums: Map[Type, Set[(String, Type)]] = root.definitions.values
+      .flatMap(x => findEnums(x.exp)).groupBy(x => x._1)
+      .mapValues(x => x.map(y => y._2).toSet)
```

Comment:
I knew about this, but I think since I immediately use declaredEnums I force the lazy val to get evaluated.
I got rid of this so we should be good either way.